presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java

```diff
@@ -876,6 +879,23 @@ public void testPartitionSchemaMismatch()
     }
 
     @Test
+    public void testPartitionSchemaNonCanonical()
+            throws Exception
+    {
+        ConnectorTableHandle table = getTableHandle(tablePartitionSchemaChangeNonCanonical);
+        ConnectorColumnHandle column = null;
+        for (Map.Entry<String, ConnectorColumnHandle> entry : metadata.getColumnHandles(table).entrySet()) {
+            if (entry.getKey().equals("t_boolean")) {
+                column = entry.getValue();
+            }
+        }
+        assertNotNull(column);
+        ConnectorPartitionResult partitionResult = splitManager.getPartitions(table, TupleDomain.withFixedValues(ImmutableMap.of(column, false)));
+        assertEquals(partitionResult.getPartitions().size(), 1, "Expected 1 partition");
+        assertEquals(partitionResult.getPartitions().get(0).getPartitionId(), "t_boolean=0", "Expected partitionId to be 't_boolean=0'");
+    }
```

Comment:
This should also test reading the partition and non-partition columns. (and if it is expected to fail, verify that and add a TODO to support it)